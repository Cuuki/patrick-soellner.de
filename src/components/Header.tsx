/** @jsxImportSource theme-ui */
import Link from 'next/link';
import { Home } from 'emotion-icons/fa-solid';
import { ColorModeToggle } from './ColorModeToggle';

export type HeaderProps = {
  siteTitle: string;
  maxWidth?: number;
};

export const Header = ({ siteTitle, maxWidth = 960 }: HeaderProps) =>  (
    <header
      sx={{
        'color': 'text',
        '@media print': {
          display: 'none',
        },
      }}
    >
      <div
        sx={{
          display: 'flex',
          alignItems: 'center',
          mx: 'auto',
          px: 3,
          py: 4,
          maxWidth,
        }}
      >
        <Link
          href="/"
          title={`Go to "${siteTitle}" cv page`}
          locale="en"
          sx={{
            'display': 'inline-flex',
            'mr': 'auto',
            'p': 1,
            'textDecoration': 'none',
            'color': 'accent',
            '&:hover': {
              color: 'accentActive',
            },
          }}
        >
          <Home size={30} />
        </Link>
        <ColorModeToggle />
      </div>
    </header>
  );
