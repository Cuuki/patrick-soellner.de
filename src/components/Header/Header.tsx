/** @jsx jsx */
import {jsx, useColorMode} from 'theme-ui';
import React, {useEffect} from 'react';
import {darken} from '@theme-ui/color';
import {Link} from 'gatsby';
import {Home} from '@emotion-icons/fa-solid';
import ColorModeToggle from '../ColorModeToggle';

const Header: React.FC<{
  siteTitle: string;
}> = ({siteTitle = ''}) => {
  const [colorMode, setColorMode] = useColorMode();

  useEffect(() => {
    const isDarkMode =
      window.matchMedia &&
      window.matchMedia('(prefers-color-scheme: dark)').matches;

    setColorMode(isDarkMode ? 'dark' : 'light');
  }, [setColorMode]);

  return (
    <header
      sx={{
        mb: 4,
        color: 'text',
      }}
    >
      <div
        sx={{
          display: 'flex',
          alignItems: 'center',
          mx: 'auto',
          px: 3,
          py: 4,
          maxWidth: 960,
        }}
      >
        <Link
          to="/"
          title={`Go to "${siteTitle}" home`}
          sx={{
            display: 'inline-flex',
            mr: 'auto',
            p: 1,
            color: 'text',
            textDecoration: 'none',
            '&:hover': {
              color: darken('text', 0.1),
            },
          }}
        >
          <Home size={30} />
        </Link>
        <ColorModeToggle mode={colorMode} handleClick={setColorMode} />
      </div>
    </header>
  );
};

export default Header;
