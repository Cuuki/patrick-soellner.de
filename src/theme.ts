import type { ColorMode, Theme } from 'theme-ui';
import { merge } from 'theme-ui';
import { deep, swiss } from '@theme-ui/presets';

const theme: Theme = {
  config: {
    useColorSchemeMediaQuery: true,
    initialColorModeName: 'dark',
    useLocalStorage: false,
  },
  colors: {
    modes: {
      light: {
        ...swiss.colors,
        accent: '#1d4ed8',
        accentActive: '#1e3a8a',
      } as ColorMode,
      dark: {
        ...deep.colors,
        accent: '#f59e0b',
        accentActive: '#b45309',
      } as ColorMode,
    },
  },
  breakpoints: ['40em', '48em', '64em', '80em'],
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
};

export default merge(deep, theme);
