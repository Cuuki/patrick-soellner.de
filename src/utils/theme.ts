import type { Theme } from 'theme-ui';
import { merge } from 'theme-ui';
import { deep, swiss } from '@theme-ui/presets';

const theme: Theme = {
  config: {
    useColorSchemeMediaQuery: true,
    initialColorModeName: 'dark',
    useLocalStorage: false,
  },
  colors: {
    modes: {
      dark: {
        ...deep.colors,
        primary: '#f59e0b',
      },
      light: {
        ...swiss.colors,
        primary: '#1d4ed8',
      },
    },
  },
  breakpoints: ['40em', '48em', '64em', '80em'],
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
};

export default merge(deep, theme);
