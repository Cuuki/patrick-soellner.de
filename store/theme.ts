import { defineStore } from 'pinia';

enum Mode {
  LIGHT = 'light',
  DARK = 'dark',
}

export const useThemeStore = defineStore('theme', {
  state: () => ({
    colorMode: Mode.LIGHT,
  }),
  actions: {
    toggleMode() {
      this.colorMode = this.colorMode === Mode.LIGHT ? Mode.DARK : Mode.LIGHT;
    },
  },
});
