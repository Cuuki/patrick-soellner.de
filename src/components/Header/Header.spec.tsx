import React from 'react';
import {render} from '@testing-library/react';
import {ThemeProvider} from 'theme-ui';
import Header from './Header';
import theme from '../../gatsby-plugin-theme-ui';

describe('Header', () => {
  it('displays the correct title', () => {
    const {container} = render(
      <ThemeProvider theme={theme}>
        <Header siteTitle="Gatsby is awesome!" />
      </ThemeProvider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
