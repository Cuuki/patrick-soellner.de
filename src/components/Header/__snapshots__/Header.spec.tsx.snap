// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header displays the correct title 1`] = `
<header
  class="css-12dkwpu-Header"
>
  <div
    class="css-obnb2a-Header"
  >
    <a
      class="css-vsikfh-Header"
      href="/"
      title="Go to \\"Gatsby is awesome!\\" home"
    >
      <svg
        aria-hidden="true"
        class="css-18cthx4-EmotionIconBase ex0cdmw0"
        fill="currentColor"
        focusable="false"
        height="30"
        viewBox="0 0 576 512"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M280.37 148.26L96 300.11V464a16 16 0 0016 16l112.06-.29a16 16 0 0015.92-16V368a16 16 0 0116-16h64a16 16 0 0116 16v95.64a16 16 0 0016 16.05L464 480a16 16 0 0016-16V300L295.67 148.26a12.19 12.19 0 00-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 00-12-12h-56a12 12 0 00-12 12v72.61L318.47 43a48 48 0 00-61 0L4.34 251.47a12 12 0 00-1.6 16.9l25.5 31A12 12 0 0045.15 301l235.22-193.74a12.19 12.19 0 0115.3 0L530.9 301a12 12 0 0016.9-1.6l25.5-31a12 12 0 00-1.7-16.93z"
          fill="currentColor"
        />
      </svg>
    </a>
    <button
      aria-label="Toggle dark mode"
      class="css-a7rtob-ColorModeToggle"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="css-18cthx4-EmotionIconBase ex0cdmw0"
        fill="currentColor"
        focusable="false"
        height="30"
        viewBox="0 0 352 512"
        width="30"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96.06 454.35c.01 6.29 1.87 12.45 5.36 17.69l17.09 25.69a31.99 31.99 0 0026.64 14.28h61.71a31.99 31.99 0 0026.64-14.28l17.09-25.69a31.989 31.989 0 005.36-17.69l.04-38.35H96.01l.05 38.35zM0 176c0 44.37 16.45 84.85 43.56 115.78 16.52 18.85 42.36 58.23 52.21 91.45.04.26.07.52.11.78h160.24c.04-.26.07-.51.11-.78 9.85-33.22 35.69-72.6 52.21-91.45C335.55 260.85 352 220.37 352 176 352 78.61 272.91-.3 175.45 0 73.44.31 0 82.97 0 176zm176-80c-44.11 0-80 35.89-80 80 0 8.84-7.16 16-16 16s-16-7.16-16-16c0-61.76 50.24-112 112-112 8.84 0 16 7.16 16 16s-7.16 16-16 16z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</header>
`;
